src/tests/ftest/cart/tests_common.h

```diff
@@ -51,11 +51,12 @@ struct test_options {
 	int		num_attach_retries;
 	bool		is_server;
 	bool		assert_on_error;
+	int		shutdown;
```

Comment:
Variable shutdown should be declared as volatile.  It may be tested/set from 2 different code streams asynchronously.

Reply:
You cant really declare volatile as part of structure. Even if one could the usage of volatile won't be correct here, instead if we were to hit issues with synchronization we would need mem barriers instead

Reply:
Member of structures may be declared as volatiles.  It is not just forcing the variable into memory, it has to do with how the compiler uses it.  If the variable is not declared as volatile, then the compiler may make a local copy of it and store it in a register or its stack.  When declared as a volatile, it forces the compiler to always go to memory to fetch the variable.

Reply:
ok